src/olympia/addons/indexers.py

```diff
@@ -231,7 +231,7 @@ def extract_document(cls, obj):
         # We can use all_categories because the indexing code goes through the
         # transformer that sets it.
         data['category'] = [cat.id for cat in obj.all_categories]
-        if obj.current_version:
+        if obj.is_listed and obj.current_version:
```

Comment:
This is a temporary hack because it's using `current_version`. Once I remove the exception in `current_version`, it will return `None` for add-ons without listed public versions, and we'll never enter that `if`.

I don't think we need to search for unlisted add-ons per platform / app version, so it should be fine.